autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -1193,6 +1338,182 @@ async def get_pending_submissions(
+            if user_ids:
+                where_dict["OR"].append({"owningUserId": {"in": user_ids}})
+
+        # Calculate pagination
+        skip = (page - 1) * page_size
+
+        # Create proper Prisma types for the query
+        where = prisma.types.StoreListingWhereInput(**where_dict)
+        include = prisma.types.StoreListingInclude(Versions=True, OwningUser=True)
+
+        # Query listings with their versions
+        listings = await prisma.models.StoreListing.prisma().find_many(
+            where=where,
+            skip=skip,
+            take=page_size,
+            include=include,
+            order=[{"createdAt": "desc"}],
+        )
+
+        # Get total count for pagination
+        total = await prisma.models.StoreListing.prisma().count(where=where)
+        total_pages = (total + page_size - 1) // page_size
+
+        # Convert to response models
+        listings_with_versions = []
+        for listing in listings:
+            versions = []
+            # Filter versions by status if specified and ensure versions is not None
+            if listing.Versions is not None:
+                filtered_versions = [
+                    v
+                    for v in listing.Versions
+                    if status is None or v.submissionStatus == status
+                ]
+
+                # Skip listings with no matching versions if a status filter is applied
+                if status and not filtered_versions:
+                    continue
+
+                for version in filtered_versions:
+                    version_model = backend.server.v2.store.model.StoreSubmission(
+                        agent_id=version.agentId,
+                        agent_version=version.agentVersion,
+                        name=version.name,
+                        sub_heading=version.subHeading,
+                        slug=listing.slug,
+                        description=version.description,
+                        image_urls=version.imageUrls or [],
+                        date_submitted=version.submittedAt or version.createdAt,
+                        status=version.submissionStatus,
+                        runs=0,  # Default values since we don't have this data here
+                        rating=0.0,  # Default values since we don't have this data here
+                        store_listing_version_id=version.id,
+                        reviewer_id=version.reviewerId,
+                        review_comments=version.reviewComments,
+                        internal_comments=version.internalComments,
+                        reviewed_at=version.reviewedAt,
+                        changes_summary=version.changesSummary,
+                        version=version.version,  # Include the actual version number
+                    )
+                    versions.append(version_model)
+
+            # Get the latest version (first in the sorted list)
+            latest_version = versions[0] if versions else None
```

Comment:
I don't see this being sorted in the db query include.

Reply:
fixed, can't be sorted in include because how prisma orderby works afaik